scripts/release-notes.py

```diff
@@ -88,17 +109,73 @@ def main(token, repo, num_commits, exclude_dependabot):
         if not pulls:
             short_sha = sha[:7]
             commit_url = commit['html_url']
-            print(f'* {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))')
+            
+            result = f'* {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))'
+            other_results.append(result)
             continue
 
         pull = pulls[0]
         pull_id = pull['number']
         pull_url = pull['html_url']
         msg = msg.replace(f'(#{pull_id})', '').strip()
-        print(f'* {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))')
+
+        # Check if the pull request has changelog label
+        pull_labels = get_pull_request_labels(token, args.repo, pull_id)
+        changelog_labels = [label for label in pull_labels if label.startswith('changelog:')]
+
+        # Handle multiple changelog labels
+        if len(changelog_labels) > 1:
+            commits_with_multiple_labels.append((sha, pull_id, changelog_labels))
+            continue
+
+        category = UNCATTEGORIZED
+        if changelog_labels:
+            for cat in categories:
+                if changelog_labels[0].startswith(cat['label']):
+                    category = cat['title']
+                    break
+            
+        result = f'* {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))'
+        if category == UNCATTEGORIZED:
+            other_results.append(result)
+        else:
+            category_results[category].append(result)
+
+    # Print categorized pull requests
+    print()
+    for category, results in category_results.items():
+        if results and category:
+            print(f'{category}:')
+            for result in results:
+                print(result)
+            print()
+
+    # Print pull requests in the 'UNCATTEGORIZED' category
+    if other_results:
+        print(f'#### {UNCATTEGORIZED}:')
+        for result in other_results:
+            print(result)
+
+    # Print warnings for commits with more than one changelog label
+    if commits_with_multiple_labels:
+        print("\nWarnings: Commits with more than one changelog label found. Please fix them:\n")
+        for sha, pull_id, labels in commits_with_multiple_labels:
+            pr_url = f"https://github.com/jaegertracing/{repo}/pull/{pull_id}"
+            print(f"Commit {sha} associated with multiple changelog labels: {', '.join(labels)}")
+            print(f"Pull Request URL: {pr_url}\n")
 
     if skipped_dependabot:
-        print(f"\n(Skipped {skipped_dependabot} dependabot commit{'' if skipped_dependabot == 1 else 's'})")
+        if not commits_with_multiple_labels:
```

Comment:
odd coupling of the logic, better to have each section to be self-contain, e.g. by always calling print() at the end

Reply:
Done 🚀